src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -182,19 +133,19 @@ public async Task<bool> Initialize(
                 replaceEnvVar: false,
                 replacementFailureMode: EnvironmentVariableReplacementFailureMode.Ignore))
         {
-            _runtimeConfig = runtimeConfig;
+            ConfigLoader.RuntimeConfig = runtimeConfig;
```

Comment:
This seems circular where 
1. The `RuntimeConfigProvider` is maintaining the `ConfigLoader`'s `RuntimeConfig` state (this line)
2. This `RuntimeConfigProvider`'s other function utilize `ConfigLoader.RuntimeConfig` state to update the state of this provider. (line 105)

Reply:
I don't follow, maybe we can sync offline and go over this one.